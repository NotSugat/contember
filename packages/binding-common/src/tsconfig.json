{
	"extends": "../../../tsconfig.settings.json",
	"compilerOptions": {
		"outDir": "../dist/types",
		"types": [
			"vite/client"
		]
	},
	"references": [
		{ "path": "../../client/src" },
		{ "path": "../../client-content/src" },
		{ "path": "../../utilities/src" },
	]
}
