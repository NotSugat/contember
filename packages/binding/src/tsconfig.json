{
	"extends": "../../../tsconfig.settings.json",
	"compilerOptions": {
		"outDir": "../dist/types",
		"types": [
			"vite/client"
		]
	},
	"references": [
		{ "path": "../../binding-common/src" },
		{ "path": "../../binding-legacy/src" },
		{ "path": "../../binding-ng/src" },
	]
}
