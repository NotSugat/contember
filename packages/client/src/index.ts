export * from './content'
export * from './system'
export * from './tenant'

export * from '@contember/client-content'
export * from '@contember/graphql-client'

export type { Input, Value, Result, Writable } from '@contember/schema'
