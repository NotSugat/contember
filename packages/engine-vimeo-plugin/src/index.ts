import { Plugin, SchemaContributorArgs } from '@contember/engine-plugins'
import { VimeoConfigProcessor } from './VimeoConfigProcessor'
import { VimeoSchemaContributor } from './VimeoSchemaContributor'
import { VimeoServiceFactory } from './VimeoService'
import { ProjectVimeoConfig } from './Config'

export * from './VimeoSchemaContributor'
export * from './VimeoService'

export default class VimeoPlugin implements Plugin<ProjectVimeoConfig> {
	name = 'contember/vimeo'

	getConfigProcessor() {
		return new VimeoConfigProcessor()
	}

	getSchemaContributor() {
		const vimeoServiceFactory = new VimeoServiceFactory()
		return new VimeoSchemaContributor(vimeoServiceFactory)
	}
}
