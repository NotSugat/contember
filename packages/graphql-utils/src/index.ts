export * from './DateTimeType'
export * from './IntervalType'
export * from './json'
export * from './JsonType'
export * from './errors'
export * from './UuidType'
