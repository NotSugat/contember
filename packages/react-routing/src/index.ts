export * from './components'
export * from './contexts'
export * from './dto'
export * from './hooks'
export * from './types'
export * from './PageNotFound'
