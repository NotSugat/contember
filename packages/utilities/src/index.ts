export * from './assert-types'
export * from './deprecate'
export * from './dom'
export * from './structures'
