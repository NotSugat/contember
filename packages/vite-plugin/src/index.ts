import { createHash } from 'node:crypto'
import { Plugin } from 'vite'

type ContemberOptions = {
	buildVersion?: boolean
	disableMiddleware?: boolean
	appPath?: string
}

export function contember(options?: ContemberOptions): Plugin {
	const appPath = options?.appPath ?? '/app'
	const contemberDsn = process.argv.find(it => it.includes('contember://'))
	const projectName = contemberDsn ? new URL(contemberDsn).username : null

	const defineConfig = projectName ? {
		'import.meta.env.VITE_CONTEMBER_ADMIN_PROJECT_NAME': JSON.stringify(projectName),
	} : {}

	return ({
		name: 'contember',
		config(config) {
			return {
				define: defineConfig,
				base: '/',
				...config,
				build: {
					sourcemap: true,
					...config.build,
					rollupOptions: {
						...config.build?.rollupOptions,
						input: config.build?.rollupOptions?.input ?? {
							root: './index.html',
							app: `.${appPath}/index.html`,
						},
					},
				},
			}
		},
		configureServer(serve) {
			if (!options?.disableMiddleware) {
				serve.middlewares.use((req, res, next) => {
					if (req.url === appPath || req.url?.startsWith(`${appPath}/`) && !req.url?.match(/\.\w+($|\?)/)) {
						req.url = `${appPath}/`
					}
					next()
				})
			}
		},
		transformIndexHtml: options?.buildVersion === false
			? undefined
			: {
				order: 'post',
				handler: (html, ctx) => {
					if (ctx.server && options?.buildVersion !== true) {
						return { html, tags: [] }
					}

					const fileHash = createHash('md5').update(html).digest().toString('hex')
					return ({
						html,
						tags: [
							{
								tag: 'meta',
								injectTo: 'head',
								attrs: {
									name: 'contember-build-version',
									content: fileHash,
								},
							},
						],
					})
				},
			},
	})
}

export default contember
