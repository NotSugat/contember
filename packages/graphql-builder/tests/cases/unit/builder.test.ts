import { describe, expect, test } from 'bun:test'
import { GraphQlField, GraphQlQueryPrinter } from '../../../src'

describe('builder', () => {

	test('fetch field', async () => {
		const printer = new GraphQlQueryPrinter()
		const result = printer.printDocument('query', [
			new GraphQlField(null, 'test', {}),
		],  {})
		expect(result).toMatchSnapshot()
	})

	test('variable deduplication', async () => {
		const printer = new GraphQlQueryPrinter()
		const someValue = { foo: 'bar' }
		const result = printer.printDocument('query', [
			new GraphQlField('b', 'test', {
				variable: { graphQlType: 'Json', value: someValue },
			}),
			new GraphQlField('a', 'test', {
				variable: { graphQlType: 'Json', value: someValue },
			}),
		],  {})
		expect(result).toMatchSnapshot()

	})
})
