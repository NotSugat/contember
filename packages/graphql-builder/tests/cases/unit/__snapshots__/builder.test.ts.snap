// Bun Snapshot v1, https://goo.gl/fbAQLP

exports[`builder fetch field 1`] = `
{
  "query": 
"query {
\ttest
}
"
,
  "variables": {},
}
`;

exports[`builder variable deduplication 1`] = `
{
  "query": 
"query($Json_0: Json) {
\tb: test(variable: $Json_0)
\ta: test(variable: $Json_0)
}
"
,
  "variables": {
    "Json_0": {
      "foo": "bar",
    },
  },
}
`;
